File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map1=new Map();
    
    for(var i=0;i<nums.length;i++){
        var b=target-nums[i];
        
        if(map1.has(b)){
            return [map1.get(b),i];
        }
        map1.set(nums[i],i);
    }
    
    return [-1,-1];
};

File: 1-two-sum/1-two-sum.go
func twoSum(nums []int, target int) []int {
    m:=make(map[int]int)
    for i:=0;i<len(nums);i++ {
        b:=target-nums[i]
        if j,ok :=m[b]; ok {
            return []int{i,j}
        }
        m[nums[i]]=i
    }
    return []int{-1,-1}
}